aws-xray-recorder-sdk-apache-http/src/main/java/com/amazonaws/xray/proxies/apache/http/TracedHttpClient.java

```diff
@@ -77,9 +78,27 @@ public static String getUrl(HttpUriRequest request) {
         return request.getURI().toString();
     }
 
-    public static String getUrl(HttpHost target, HttpRequest request) {
-        return target.getHostName() + request.getRequestLine().getUri();
-    }
+	public static String getUrl(HttpHost target, HttpRequest request)
+	{
+		URI requestURI = null;
+		try {
+			requestURI = new URI(request.getRequestLine().getUri());
+		}
+		catch (URISyntaxException e) {
+			// Not a valid URI, so it's obviously not absolute. Default to
+			// hostname + request line
+		}
+
+		// Check if the uri in the request line is an absolute or relative url.
+		// If it's aboslute, just use it as the url. If it's not, concatenate
+		// the hostname with the relative path to (almost) get an absolute url.
+		if(requestURI != null && requestURI.isAbsolute()) {
```

Comment:
instead of doing a null check here (and have requestURI at the method scope) it seems this case can be handled within the try block, i.e.

```java
String uri = request.getRequestLine().getUri();

try {
    URI requestURI = new URI(uri);
    if (requestURI.isAbsolute()) {
        return requestURI.toString();
    }
} catch (...) {
    // Not a valid ...
}

return target.getHostName() + uri;
```